### ISSUE TITLE
A 6st & 7th step in the workflow

### ISSUE THREAD

**Issue by emielvanloon** [zoonproject/zoon#318]
Shouldn't we make a 6th and 7th step in the workflow (both optional, and after the current steps), to deal with (external validation) separately & potentially the fact that after the model-selection a final model could be selected & applied?

- 6 -- validate (using final model - out of the process module)
- 7 -- output (similar to the 'output' module in step 5, but the output in 7 would typically produce different items).

Taking validation serious (and out of process) may be important to present the method in a scientific paper.

**Comment by AugustT** [comment]
I may not understand this correctly but I guess the way I see this scenario going is;l I use zoon to run a bunch of different model modules and to compare them

```r
myresult <- workflow(mydata,
                     myhabitatmap,
                     background(),
                     list(model1,model2,model3),
                     PermormanceMeasures)
```

Then once I have selected my model I run a 'production run'

```r
myresult <- workflow(mydata,
                     myhabitatmap,
                     background(),
                     model1,
                     PrintMap)
```

I don't think that we have to be able to do the entire scientific process in one run of the workflow function.

**Comment by timcdlucas** [comment]
I guess the difficulty with this is that the whole model selection process does have implications for interpreting the final model. If I fitted a million different models, one will do well on external data by chance. If I fit models with different variables, I can't then write up my methods as though I selected the variables a priori, nor can I usefully use many statistical tests on those variables.

That said, 7 steps is certainly getting very confusing... 

If we get the `Aggregate` function going, I think we can deal with some of these issues. For example, refitting and validating the best model would become an output module based on aggregating a number of model modules.
```
myresult <- workflow(mydata,
                     myhabitatmap,
                     background(),
                     list(model1,model2,model3),
                     list(PerformanceMeasures, Aggregate(PredictFromBestModel)))
```

Or something. I'm getting myself confused.
